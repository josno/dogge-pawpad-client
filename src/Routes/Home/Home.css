.home-page-title {
	color: #fcfcfc;
	font-family: Candal, sans-serif;
	font-size: 3em;
	padding-top: 60px;
	margin: 0px;
	width: 100%;
	-webkit-text-stroke: 1px black;
}

.intro-details,
.header-title {
	align-items: center;
	display: flex;
	flex-direction: column;
	justify-content: center;
}

.intro-details {
	padding: 20px;
}

.header-title {
	background-color: #009fb7;
}

.homepage {
	height: 100%;
}

.homepage-image {
	object-position: bottom 0px right 30px;
	width: 300px;
	height: 200px;
}

.options {
	margin-left: auto;
	margin-right: auto;
	height: fit-content;
	text-align: center;
	vertical-align: middle;
}

.form-tab {
	border: 2px solid #99e1d9;
	box-shadow: inset 0 0 0 0 #99e1d9;
	color: #fff;
	cursor: pointer;
	display: inline-block;
	font-weight: bold;
	height: 80px;
	letter-spacing: 1px;
	line-height: 60px;
	margin: 20px;
	padding: 10px;
	text-align: center;
	transition: ease-out 0.4s;
	vertical-align: middle;
	width: 80vw;
	-webkit-transition: ease-out 0.4s;
	-moz-transition: ease-out 0.4s;
}

.form-tab:hover {
	box-shadow: inset 800px 0 0 0 #99e1d9;
}

a:hover {
	text-shadow: 2px 3px #ff7300;
}

.input-login-button {
	border: 2px solid black;
	width: 40vh;
	margin: 10px auto;
	font-size: 1em;
}

.input-login-button:hover {
	cursor: pointer;
}

@media (min-width: 700px) {
	.form-tab {
		width: 400px;
	}

	.intro-details {
		display: flex;
	}

	.options {
		display: flex;
		flex-direction: column;
	}

	.header-title {
		flex-direction: row;
		padding-top: 20px;
	}

	.home-page-title {
		font-size: em;
		left: 5vw;
		padding-top: 0;
		position: relative;
		width: 40vw;
	}

	.homepage-image {
		height: 250px;
		right: -30px;
		width: 350px;
		object-position: bottom 0px right 30px;
		position: relative;
	}
}

@media (min-width: 1200px) {
	.header-title {
		flex-direction: row;
		justify-content: center;
		align-items: center;
		padding-top: 20px;
	}

	.homepage-image {
		left: 0;
		width: 400px;
	}

	.home-page-title {
		font-size: 4em;
		width: 40%;
	}
}
