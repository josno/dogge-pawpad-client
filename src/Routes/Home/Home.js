import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import DogImage from '../../assets/ken-reid-Cnrxu_Za30M-unsplash-removebg-preview.png';
import PawPadContext from '../../PawPadContext';
import TokenService from '../../services/token-service';
import './Home.css';

class Home extends Component {
	static contextType = PawPadContext;
	render() {
		return (
			<main className="homepage">
				<header className="header-title" role="banner">
					<h1 className="home-page-title"> PawPad </h1>
					<img
						className="homepage-image"
						src={DogImage}
						alt="dog"
					></img>
				</header>
				<section className="intro-details">
					<h3> An animal shelter intake application.</h3>

					<p>
						Check on your doggos and update essential information.
						Stay more organized and stay on track - doggo
						appreciates it!
					</p>
				</section>

				{TokenService.hasAuthToken() ? (
					<section>
						<div className="options">
							<ul>
								<li className="form-tab">
									<Link to={'/dogs-list'}>See Dogs</Link>
								</li>

								<li className="form-tab">
									<Link to={'add-new-dog'}>Add A Dog</Link>
								</li>
							</ul>
						</div>
					</section>
				) : (
					<section className="form-tab">
						<Link to={'login'}>Login To See Dogs</Link>
					</section>
				)}
			</main>
		);
	}
}

export default Home;
