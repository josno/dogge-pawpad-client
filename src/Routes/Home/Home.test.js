import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from './Home';

describe('<Home />', () => {
	it('renders as expected', () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<Home />
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
