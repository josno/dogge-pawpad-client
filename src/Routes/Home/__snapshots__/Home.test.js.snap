// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders as expected 1`] = `
<main
  className="homepage"
>
  <header
    className="header-title"
    role="banner"
  >
    <h1
      className="home-page-title"
    >
       PawPad 
    </h1>
    <img
      alt="dog"
      className="homepage-image"
      src="ken-reid-Cnrxu_Za30M-unsplash-removebg-preview.png"
    />
  </header>
  <section
    className="intro-details"
  >
    <h3>
       An animal shelter intake application.
    </h3>
    <p>
      Check on your doggos and update essential information. Stay more organized and stay on track - doggo appreciates it!
    </p>
  </section>
  <section
    className="form-tab"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      Login To See Dogs
    </a>
  </section>
</main>
`;
